Sources/muterCore/Extensions/StringExtensions.swift

```diff
@@ -67,3 +67,31 @@ extension String {
         return String(self[range])
     }
 }
+
+/// Converts a UTF-8 byte offset to a character offset in the given string.
+///
+/// - Parameters:
+///   - string: The string in which the offset needs to be converted.
+///   - utf8Offset: The UTF-8 byte offset to be converted.
+/// - Returns: The corresponding character offset if the conversion is successful, or `nil` if the offset is out of range.
+///
+/// - Complexity: O(*n*), where *n* is the number of characters in the string.
+func convertUtf8OffsetToCharOffset(in string: String, at utf8Offset: Int) -> Int? {
```

Comment:
2 minor things to make this more aesthetic.
if you move this to the `String` extension, we could do this

```swift
codeBlockDescription.convertUtf8OffsetToCharOffset(at: utf8Offset)
```

I would suggest naming it to something link `fixOffsetIfNeeded(_ offset: Int)`

Also, instead of returning `nil`,  we could return the `offset` that was passed by the call site, this way there is no need to `guard` anything.

Reply:
@rakaramos  I think the name `fixOffsetIfNeeded` can not explain the context without some comments